### PR TITLE
Added 2020 census population entities

### PR DESCRIPTION
**Pull request by thy-neighbor**

### CODE REVIEW

File: common/constants.py [hunk 1]
```diff
@@ -43,6 +43,9 @@ class EntityKey:
     childcare_center_type = 'childcare.center_type'
     wholesale_market_app_type = 'wholesale.application_type'
     wholesale_market = 'wholesale.market'
+    population_zipcode_2020 = 'population.zipcode_2020'
```
Comment (CThaw90) [inline common/constants.py]: Follow naming convention for the other census population data we have in this file
Ex: `census_city_population`

**Comment by CThaw90** [comment]
Move the new population modules into their corresponding named files in the census package

population/blockgroup2020 -> census/blockgroup 
population/censustract2020 -> census/tract
population/zipcode2020 -> census/zipcode

Follow pattern for city population module in the census package

File: census/tract.py [hunk 2]
```diff
@@ -16,6 +16,9 @@ class CensusTract(ResourceEntity):
     @staticmethod
     def dependencies():
         return [entity_key.census_us_county]
+    
+    def format_census_tract(self, subject_tract):
```
Comment (CThaw90) [inline census/tract.py]: This method can be static since you're not using the self reference

File: population/zipcode2020.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    def fetch(self):
+        self.records = []
+        PAGEABLE_MAX = 56
+        duplicate_set = set()
+        
+        for i in range(PAGEABLE_MAX):
+            i += 1
+            
+            if(i in ({3,7,14,43,52})):
```
Comment (CThaw90) [inline population/zipcode2020.py]: What are these? Indexes? Can we put this structure in a named variable that describes what these numbers are?

File: population/zipcode2020.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            )
+            
+            response = json.loads(requests.request('GET', request_url).content.decode('utf-8'))
+            data = response['response']['data']
+            # Note: the first element is improper
+            data.pop(0)
+            for record in data:
+                if record[5] not in duplicate_set:
+                    duplicate_set.add(record[5])
+                    self.records.append({'population': record[2], 'zipcode': record[5]})
```
Comment (CThaw90) [inline population/zipcode2020.py]: We should try to avoid magic numbers. Create a variable for `5` and `2`.

File: population/blockgroup2020.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            records = self.mysql_client.select(self.table_name)
+            block_group_entity = self.dependencies_map[entity_key.census_block_group]
+            
+            for record in records:
+                for field in self.cacheable_fields:
+                    block_group_record = block_group_entity.get_cached_value(record[field])
+                    formatted_block_group = block_group_entity.format_block_group(block_group_record['name'])
+                    self.record_cache[formatted_block_group] = record
+
+    def fetch(self):
```
Comment (CThaw90) [inline population/blockgroup2020.py]: This seems like the same function outside of the URL and geographic attribute. Is it possible to create a reusable function that takes in the API_URL and attribute (ex: `block_group`) as parameters and reuse it for block_group, census_tract and zipcode?

File: census/tract.py [hunk 6]
```diff
@@ -32,11 +35,24 @@ def __init__(self):
             {'field': 'code', 'column': 'county_id', 'data': self.get_county_id}
         ]
 
-        self.cacheable_fields = ['fips']
+        self.cacheable_fields = ['fips','name','id']
```
Comment (CThaw90) [inline census/tract.py:40]: Do we need to cache by `id` also? What is using that?

Comment (thy-neighbor) [reply census/tract.py:40]: It's used in TractPopulation -> load_cache() to identify the name using the id. Which is then used for the skipping logic

**Review by CThaw90 — APPROVED** [review]

**Pull request merged**